packages/@lightningjs/ui-components/src/components/Control/Control.d.ts

```diff
@@ -17,26 +17,63 @@
  */
 
 import type lng from '@lightningjs/core';
-import { ButtonSmall, ButtonStyle } from '../Button';
-import type { StylePartial } from '../../types/lui';
+import ButtonSmall, { ButtonSmallStyle } from '../Button/ButtonSmall';
+import Button from '../Button';
+import { StylePartial } from '../../types/lui';
 
 type LogoStyleObject = {
   radius: lng.Tools.CornerRadius;
   h: number;
   w: number;
 };
 
-export type ControlStyle = ButtonStyle & {
-  h: number;
+type ControlStyle = ButtonSmallStyle & {
   iconStyle: Record<string, unknown>;
   logoStyle: LogoStyleObject;
   radius: lng.Tools.CornerRadius;
 };
 
-export default class Control extends ButtonSmall {
+declare namespace Control {
+  // TODO: need to replace Button with ButtonSmall here , but throws error. Is this correct now?
+  export interface TemplateSpec extends Button.TemplateSpec {
+    /**
+     *  URL for the icon asset
+     */
+    icon?: string;
+
+    /**
+     *  URL for the icon asset
+     */
+    logo?: string;
+
+    /**
+     * flag that if `true`, hides the title when `Control` is in unfocused or disabled mode
+     */
+    shouldCollapse?: boolean;
+  }
+}
+
+declare class Control<
+  TemplateSpec extends Control.TemplateSpec = Control.TemplateSpec,
+  TypeConfig extends lng.Component.TypeConfig = lng.Component.TypeConfig
```

Comment:
@anthony9187 Still trying to make sure I understand this correctly. Should this TypeConfig be extending the Button's TypeConfig instead, or is this correct?

Reply:
this is correct since Button doesn't have a TypeConfig and itself uses lng.Component's TypeConfig